### PR TITLE
added check to avoid IR generation in case of wrong input shape

### PR DESCRIPTION
**Pull request by sadolini**
Description: added check to avoid IR generation in case of wrong input shape

JIRA: 35684

Code:
* [x]  Comments **NA too simple change** 
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR **NA no new transformations**
* [x]  Transformation preserves original framework node names **NA no new transformations**

Validation:
* [x]  Unit tests
* [x]  Framework operation tests **NA no new operations**
* [x]  Transformation tests **NA no new transformations**
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py **NA change covered unit test**
* [x]  Model Optimizer IR Reader check **NA no new operations**

Documentation:
* [x]  Supported frameworks operations list **NA no new operations**
* [x]  Supported **public** models list **NA no new models**
* [x]  New operations specification **NA no new operations**
* [x]  Guide on how to convert the **public** model **NA no new models**
* [x]  User guide update **NA nothing new for user**

### CODE REVIEW

File: tests/time_tests/common/main.cpp [hunk 1]
```diff
@@ -27,12 +27,21 @@ bool parseAndCheckCommandLine(int argc, char **argv) {
 }
 
 
+/**
+* @brief Function calls `runPipeline` with mandatory time tracking of full run
+*/
+int _runPipeline() {
```
Comment (vurusovs) [inline tests/time_tests/common/main.cpp:33]: looks like the code have been added after some conflicts resolving and shouldn't be in the PR

File: model-optimizer/mo/ops/pooling.py [hunk 2]
```diff
@@ -96,9 +94,14 @@ def infer(node: Node):
...
             if node.soft_get('pooling_convention') == 'full' or node.soft_get('rounding_type') == 'ceil':
                 rounding = np.ceil
-            output_spatial_shape = np.array(rounding(
-                np.array(input_spatial_shape + pad_spatial_shape - window_spatial_shape,
-                         dtype=np.float) / stride_spatial), dtype=np.int64) + 1
+
+            padded_spatial_shape = input_spatial_shape + pad_spatial_shape - window_spatial_shape
+            if np.any(padded_spatial_shape < 0):
+                raise Error("Data after padding has dimension less than window size. " +
+                            "Possible reason of error in incorrect input_shape")
```
Comment (lazarevevgeny) [inline model-optimizer/mo/ops/pooling.py:101]: `error in incorrect input_shape` -> `error is incorrectly specified model input shape(s).`

File: model-optimizer/mo/ops/pooling.py [hunk 3]
```diff
@@ -96,9 +94,14 @@ def infer(node: Node):
...
+
+            output_spatial_shape = np.array(rounding(np.array(padded_spatial_shape, dtype=np.float) / stride_spatial),
```
Comment (lazarevevgeny) [inline model-optimizer/mo/ops/pooling.py:103]: Use methods `int64_array` and `float_array`

File: model-optimizer/mo/ops/convolution.py [hunk 4]
```diff
@@ -72,6 +71,11 @@ def calc_convolution(input_spatial_shape, stride_spatial_shape, pad_spatial_shap
             Verified to be applicable for both Caffe and ONNX.
         '''
         spatial_val_wo_stride = input_spatial_shape + pad_spatial_shape - kernel_extent
+
+        if np.any(spatial_val_wo_stride < 0):
+            raise Error("Data after padding has dimension less than window size. " +
+                        "Possible reason of error in incorrect input_shape")
```
Comment (lazarevevgeny) [inline model-optimizer/mo/ops/convolution.py:77]: `error in incorrect input_shape` -> `error is incorrectly specified model input shape(s).`

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/mo/ops/pooling.py (hunk 2 above)
Comment (sadolini) [reply model-optimizer/mo/ops/pooling.py]: fixed

File: model-optimizer/mo/ops/convolution.py (hunk 4 above)
Comment (sadolini) [reply model-optimizer/mo/ops/convolution.py]: fixed

File: model-optimizer/mo/ops/pooling.py (hunk 3 above)
Comment (sadolini) [reply model-optimizer/mo/ops/pooling.py]: fixed

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**